src/Spago/Command/Bundle.purs

```diff
@@ -47,38 +53,85 @@ run = do
       BundleBrowser, BundleApp -> "--format=iife"
       _, _ -> "--format=esm"
 
-    -- See https://github.com/evanw/esbuild/issues/1921
-    nodePatch = case opts.platform of
-      BundleNode -> [ "--banner:js=import __module from \'module\';import __path from \'path\';import __url from \'url\';const require = __module.createRequire(import.meta.url);const __dirname = __path.dirname(__url.fileURLToPath(import.meta.url));const __filename=new URL(import.meta.url).pathname" ]
-      _ -> []
+    onlyForNode s = case opts.platform of
+      BundleNode -> s
+      BundleBrowser -> ""
 
-    output = case workspace.buildOptions.output of
-      Nothing -> "output"
-      Just o -> o
+    output = workspace.buildOptions.output # fromMaybe "output"
     -- TODO: we might need to use `Path.relative selected.path output` instead of just output there
     mainPath = withForwardSlashes $ Path.concat [ output, opts.module, "index.js" ]
 
-    shebang = case opts.platform of
-      BundleNode -> "#!/usr/bin/env node\n\n"
-      _ -> ""
-
     { input, entrypoint } = case opts.type of
-      BundleApp -> { entrypoint: [], input: Cmd.StdinWrite (shebang <> "import { main } from './" <> mainPath <> "'; main();") }
-      BundleModule -> { entrypoint: [ mainPath ], input: Cmd.StdinNewPipe }
+      BundleApp ->
+        { entrypoint: []
+        , input: Cmd.StdinWrite $ fold [ onlyForNode "#!/usr/bin/env node\n\n", "import { main } from './", mainPath, "';main();" ]
+        }
+      BundleModule ->
+        { entrypoint: [ mainPath ]
+        , input: Cmd.StdinNewPipe
+        }
+
     execOptions = Cmd.defaultExecOptions { pipeStdin = input }
 
-    args =
-      [ "--bundle"
-      , "--outfile=" <> outfile
-      , "--platform=" <> show opts.platform
-      -- See https://github.com/evanw/esbuild/issues/1051
-      , "--loader:.node=file"
-      , format
-      ] <> opts.extraArgs <> minify <> sourceMap <> entrypoint <> nodePatch
+    banner = fold
+      [ spagoGeneratedFilePrefix
+      , " v"
+      , BuildInfo.packages."spago-bin"
```

Comment:
I am confused why this version is 0 in the fixtures

Reply:
Ah I think it's because the "packages" field will contain the version for the packages being compiled. None of the packages in the fixtures is called "spago-bin", so it comes out as 0.

Since we want to print Spago's version in here, I think we should add a `spagoVersion` field in the `BuildInfo` (I was entirely convinced it was there already). Probably in a separate PR.

Reply:
It doesn't "come out" as 0. I manually replaced it with 0.

The problem is, it's supposed to be the "current" version of Spago, which means it can't be any particular version. The current version has to be inserted there during the test run, as that's the only way to get the "current" version. So the "v0" bit is just a placeholder, and it's replaced [here](https://github.com/purescript/spago/pull/1260/files#diff-d4d0ded0a864b48583b9f1fdb5c79023288c9df5fc4bd4499b8d1233b78b5dd9R73-R82), as the comment says.

Reply:
Oh yeah makes sense - thanks for the clarification!